icons/src/fullcolor/render-bitmaps-verbose.py

```diff
@@ -238,60 +221,278 @@ def characters(self, chars):
+    print("│   output directory (by default '../../Suru')     │")
+    print("│                                                  │")
+    print("│ --scaling_factors [SCALING_FACTORS]              │")
+    print("│                                                  │")
+    print("│   scaling factors to render at                   │")
+    print("│   (by default [1, 2], e.g. 100% & 200%)          │")
+    print("│                                                  │")
+    print("│ --verbose                                        │")
+    print("│                                                  │")
+    print("│   print verbose output to the terminal           │")
+    print("│                                                  │")
+    print("└──────────────────────────────────────────────────┘")
+    print("")
+    return
+
+
+def print_categories():
 
-if args.svg is None:
-    print("\nNo arguments provided; processing listed sources:\n")
-    for source in CATEGORIES_TO_RENDER:
-        print("-- \"" + os.path.join('.', source) + "\"")
     print("")
-else:
     print("┌────────────────────────────────┐")
-    print("│ Rendering from command-line argument \"" + args.svg + "\"")
+    print("│ Icon Categories:               │")
     print("├────────────────────────────────┤")
+    print("│ - actions                      │")
+    print("│ - apps                         │")
+    print("│ - categories                   │")
+    print("│ - devices                      │")
+    print("│ - emblems                      │")
+    print("│ - legacy                       │")
+    print("│ - mimetypes                    │")
+    print("│ - places                       │")
+    print("│ - status                       │")
+    print("│ - wip                          │")
+    print("└────────────────────────────────┘")
+    print("")
+    return
+
+
+def build_parser():
+
+    parser = argparse.ArgumentParser(add_help=False)
+
+    parser.add_argument(
+        '--categories',
+        type=str,
+        nargs='?',
+        default=(
+            "actions",
+            "apps",
+            "categories",
+            "devices",
+            "emblems",
+            "legacy",
+            "mimetypes",
+            "places",
+            "status",
+            "wip",
+        ),
+        help="categories of icons to render (by default all)"
+    )
+    parser.add_argument(
+        '--base_dpi',
+        type=int,
+        nargs=1,
+        default=96,
+        help="dpi to use for rendering (by default 96)"
+    )
+    parser.add_argument(
+        '--filter',
+        type=str,
+        nargs=1,
+        help="Inkscape filter to apply while rendering (by default none)"
+    )
+    parser.add_argument(
+        '--help',
+        action='store_true',
+        help="show this help message and exit"
+    )
+    parser.add_argument(
+        '--inkscape_path',
+        type=str,
+        nargs=1,
+        default = '/usr/bin/inkscape',
```

Comment:
I do not think there is really the need to use default paths for inkscape (same thing for optipng). Since Python 3.3 shutil library has a `which` implementation, that you can use to verify whether the program is installed or not.

https://docs.python.org/3.9/library/shutil.html#shutil.which

Reply:
Thanks for the link. I’ll add in the `which`, but I will probably leave in the arguments as a backup for if `which` doesn’t work or the user otherwise wants to call a custom executable.